c10/util/SmallBuffer.h

```diff
@@ -32,6 +32,25 @@ class SmallBuffer {
     }
   }
 
+  SmallBuffer(const SmallBuffer&) = delete;
+  SmallBuffer& operator=(const SmallBuffer&) = delete;
+
+  // move constructor is needed in function return
+  SmallBuffer(SmallBuffer&& rhs) : size_{rhs.size_} {
+    rhs.size_ = 0;
+    if (size_ > N) {
+      data_ = rhs.data_;
+      rhs.data_ = nullptr;
+    } else {
+      for (size_t i = 0; i < N; i++) {
+        storage_[i] = rhs.storage_[i];
```

Comment:
```suggestion
        storage_[i] = std::move(rhs.storage_[i]);
```
This will allow it to work if T is not copyable.